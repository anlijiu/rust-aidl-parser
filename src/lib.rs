mod ast;
mod error;
pub mod types;

pub fn parse(inputs: &[&str]) -> Result<Vec<types::File>, error::ParseError> {
    let files = inputs.iter().map(|i| {
        let lookup = line_col::LineColLookup::new(i);
        ast::rules::file(i, &lookup).map_err(error::ParseError::Peg)
    });

    files.collect()
}
