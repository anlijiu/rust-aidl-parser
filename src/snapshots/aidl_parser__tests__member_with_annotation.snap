---
source: src/lib.rs
expression: res

---
Member(
  name: "memberName",
  member_type: Type(
    name: "TypeName",
    kind: Custom,
    generic_types: [],
    definition: None,
    symbol_range: "...",
  ),
  value: Some("\"member value\""),
  symbol_range: "...",
  doc: None,
  annotations: [
    Annotation(
      name: "@AnnotationName",
      key_values: {},
    ),
  ],
  full_range: "...",
)
